javascript/react/redux-tips/avoid-object-mutations.md

```diff
@@ -30,15 +30,35 @@ aspects:
 
 A **pure** function that returns a new object with the needed modifications can easily be implemented in **Redux**. However, if other properties are added later on to the object, you might need to update several pieces of your code.
 
-To avoid this, the `Object.assign()` method introduced in **ES6** can be used because it returns a new object with all enumerable properties of the *source* object:
+To avoid this, the `Object.assign()` method can be used because it returns a new object with all enumerable properties of the *source* object:
+
+```jsx
+const target = { a: 1, b: 2 };
+const source = { b: 3, c: 4 };
+
+const returnedTarget = Object.assign(
+  target,
+  source
+);
+
+console.log(target);
+// { a: 1, b: 3, c: 4 }
+
+console.log(returnedTarget);
+// { a: 1, b: 3, c: 4 }
+```
+
+This method is used to create a new object by copying the properties from the source object to the target object. This is useful for Redux as the method returns a new object, which keeps in line with creating **pure** functions. When used inside `reducers`, the target object is the one whose properties are going to be assigned. Because we want to keep our function **pure** we will pass an empty object `{ }`. The source object contains the properties that are going to be copied, and we can even add new properties:
```

Comment:
Can we use the word "reducer" instead of "function" in this paragraph?

```suggestion
This method is used to create a new object by copying the properties from the source object to the target object. This is useful for Redux as the method returns a new object, which keeps in line with creating **pure** reducers. Within a reducer, the target object is the one on which we assign properties. Because we want to keep our reducer **pure** (avoid mutation) we will pass an empty object `{ }` as the starting target. The source object contains the properties that are going to be copied to the target, and we can even add new properties:
```